specification/trace/semantic_conventions/http.md

```diff
@@ -194,6 +194,30 @@ Following attributes MUST be provided **at span creation time** (when provided a
 
 Note that in some cases host and port identifiers in the `Host` header might be different from the `server.address` and `server.port`, in this case instrumentation MAY populate `Host` header on `http.request.header.host` attribute even if it's not enabled by user.
 
+### HTTP client span duration
+
+HTTP client spans SHOULD start sometime before the first request byte is sent. This may or may not include connection time.
```

Comment:
I suggest something different instead:
The HTTP span SHOULD end whenever it is natural in the context of the instrumentation (instrumented library).

Also note that any given client call may result in several HTTP spans (redirects, retries, reauthentication, ...), any of which may have the body read or not. But typically, it should IMHO end as soon as the (appropriately lower/higher-level) client function call returns. I think we must be careful not to overspecify here.

Keeping the HTTP client span open after that is problematic, e.g. the client may only read the body after some non-HTTP-related processing has been done and throw the result away to facilitate connection reuse. This will include all business logic that happened in-between in the HTTP span time.

Reply:
@Oberon00 I agree with what you wrote, but struggling to make this clear in the PR, can you make a specific suggestion? Thx

Reply:
I think the current version is already much better than the version I actually commented on (where this line was saying something different). So I think what is written here is fine.

Reply:
Maybe you could just literally add "The HTTP span SHOULD start and end whenever it is natural in the context of the instrumented library (e.g. depending on the API or inner workings). There are some minimal constraints that SHOULD usually be honored:".

Reply:
> whenever it is natural in the context of the instrumented library

I struggled with this, "natural" feels maybe too vague.

> There are some minimal constraints that SHOULD usually be honored

I like this and incorporated it.